PeakNetwork/Core/API.swift

```diff
@@ -0,0 +1,148 @@
+//
+//  API.swift
+//  PeakNetwork-iOS
+//
+//  Created by Sam Oakley on 12/03/2019.
+//  Copyright © 2019 3Squared. All rights reserved.
+//
+
+import Foundation
+
+/// A type that represents a web API.
+public protocol API {
+    
+    /// The base URL of the API. Paths will be appended to this.
+    var baseURL: String { get }
+    
+    /// The session in which to perform generated requests.
+    var session: Session { get }
+    
+    /// Common query items that will be added to every request.
+    var commonQueryItems: [String: String] { get }
```

Comment:
This should be `[String: String?]` because `URLQueryItem` can take nil for its value.